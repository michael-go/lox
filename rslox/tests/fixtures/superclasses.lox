class A {
  method() {
    print "A method";
  }
}

class B < A {
  method() {
    print "B method";
  }

  test() {
    super.method();
  }
}

class C < B {}

C().test();

print("/////");

class X {
  method() {
    print "X";
  }
}

class Y < X {
  method() {
    var closure = super.method;
    closure(); // Prints "X".
  }
}

Y().method();
