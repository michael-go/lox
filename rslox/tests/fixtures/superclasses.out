# exit code: 0
# stdout:
"A method"
"/////"
"X"

# stderr:

